crates/sparrow-runtime/src/execute/operation/shift_to.rs

```diff
@@ -451,8 +449,8 @@ impl ShiftToColumnOperation {
             let group_indices: &UInt32Array = downcast_primitive_array(group_indices.as_ref())?;
             // TODO: We shouldn't need to clone the array. Instead, we should be able
             // to pass an owned reference to the grouping indices. But... the clone
-            // shouldn't be too expensive, and that would require API hanges.
-            let group_indices: UInt32Array = UInt32Array::from(group_indices.data().clone());
+            // shouldn't be too expensive, and that would require API changes.
+            let group_indices: UInt32Array = UInt32Array::from(group_indices.to_data());
```

Comment:
Does the comment still apply?

Reply:
Yes it does -- `to_data` creates a copy of the data, and we create a new `UInt32Array` from it. It shouldn't be too expensive, but it there should still be possibilities of using an `ArrayRef` or something if it *was* expensive.